async/src/main/java/org/odk/collect/async/TaskSpecWorker.kt

```diff
@@ -61,12 +61,12 @@ class TaskSpecWorker(
 
         try {
             val completed =
-                taskSpec.getTask(applicationContext, stringInputData, isLastUniqueExecution(taskSpec)) { isStopped }.get()
+                taskSpec.getTask(applicationContext, stringInputData, foreground || isLastUniqueExecution(taskSpec)) { isStopped }.get()
             val maxRetries = taskSpec.maxRetries
 
             return if (completed) {
                 Result.success()
-            } else if (maxRetries == null || runAttemptCount < maxRetries) {
+            } else if (!foreground && (maxRetries == null || runAttemptCount < maxRetries)) {
```

Comment:
We don't want to `retry` when running in the foreground: it doesn't fit with the idea of having more immediate results/feedback, and it also could cause errors (foreground services should not be started in the background).